Mathlib/Order/Interval/Finset/Nat.lean

```diff
@@ -345,11 +345,15 @@ theorem range_image_pred_top_sub (n : ℕ) :
     simp_rw [succ_sub_succ, tsub_zero, tsub_self]
 #align finset.range_image_pred_top_sub Finset.range_image_pred_top_sub
 
--- Porting note: had to use `@` and specify `(a + b)` explicitly. mathlib3 managed without.
 theorem range_add_eq_union : range (a + b) = range a ∪ (range b).map (addLeftEmbedding a) := by
   rw [Finset.range_eq_Ico, map_eq_image]
-  convert (@Ico_union_Ico_eq_Ico _ _ _ _ _ (a + b) a.zero_le le_self_add).symm
-  exact image_add_left_Ico _ _ _
+  convert (Ico_union_Ico_eq_Ico a.zero_le le_self_add).symm
+  ext x
+  simp only [Ico_zero_eq_range, mem_image, mem_range, addLeftEmbedding_apply, mem_Ico]
+  constructor
+  · aesop
+  · rintro h
+    exact ⟨x - a, by omega⟩
```

Comment:
This doesn't look great. What's the problem here?

Reply:
`image_add_left_Ico` is an algebraic order lemma. I'm just reproving it quickly in the special case of `Nat`.